File: app/src/main/java/com/example/myanimedata/adapter/GenreAdapter.java
package com.example.myanimedata.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.myanimedata.R;
import com.example.myanimedata.api.GenreResult;

import java.util.List;

public class GenreAdapter extends RecyclerView.Adapter<GenreAdapter.GenreViewHolder>{
    private final List<GenreResult> genreResults;

    public GenreAdapter(List<GenreResult> genreResults) {
        this.genreResults = genreResults;
    }


    @NonNull
    @Override
    public GenreAdapter.GenreViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new GenreAdapter.GenreViewHolder(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.container_genre, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull GenreAdapter.GenreViewHolder holder, int position) {
        holder.bindItem(genreResults.get(position));
    }

    @Override
    public int getItemCount() {
        return genreResults.size();
    }

    public static class GenreViewHolder extends RecyclerView.ViewHolder{
        private final TextView textGenres;

        public GenreViewHolder(@NonNull View itemView) {
            super(itemView);

            textGenres = itemView.findViewById(R.id.textGenres);
        }

        public void bindItem(GenreResult genreResult){
            textGenres.setText(genreResult.getName());
        }
    }
}

File: app/src/main/java/com/example/myanimedata/adapter/GenreAdapter.kt
package com.example.myanimedata.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.myanimedata.R
import com.example.myanimedata.adapter.GenreAdapter.GenreViewHolder
import com.example.myanimedata.api.GenreResult
import com.google.android.material.chip.Chip

class GenreAdapter(private val genreResults: List<GenreResult>) :
    RecyclerView.Adapter<GenreViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): GenreViewHolder {
        return GenreViewHolder(
            LayoutInflater.from(parent.context)
                .inflate(R.layout.container_genre, parent, false)
        )
    }

    override fun onBindViewHolder(holder: GenreViewHolder, position: Int) {
        holder.bindItem(genreResults[position])
    }

    override fun getItemCount(): Int {
        return genreResults.size
    }

    class GenreViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val textGenres: Chip

        init {
            textGenres = itemView.findViewById(R.id.textGenres)
        }

        fun bindItem(genreResult: GenreResult) {
            textGenres.text = genreResult.name
        }
    }
}